File: content/problems/0621-task-scheduler/solutions/frequency-frame/Solution.java
class Solution {
    public int leastInterval(char[] tasks,int n){
        int[] counts=new int[26];
        for(char task:tasks)counts[task-'A']++;
        int maximum=0,kinds=0;
        for(int count:counts)if(count>maximum){
            maximum=count;
            kinds=1;
        }else if(count==maximum)kinds++;
        return Math.max(tasks.length,(maximum-1)*(n+1)+kinds);
    }
}

File: content/problems/0621-task-scheduler/solutions/frequency-frame/Solution.kt
class Solution {
    fun leastInterval(tasks:CharArray,n:Int):Int{
        val counts=IntArray(26);
        for(task in tasks)counts[task-'A']++;
        val maximum=counts.maxOrNull()?:0;
        val kinds=counts.count{
            it==maximum
        };
        return maxOf(tasks.size,(maximum-1)*(n+1)+kinds)
    }
}
